### PR TITLE
VI-626 Add mhv_account_creation_api flipper

### PR DESCRIPTION
**Pull request by emilykim13**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- *This work is behind a feature toggle (flipper): YES*
- *We are adding a a feature toggle `mhv_account_creation_api` in the BE to begin the phase rollout for the account creation api.*

## Related issue(s)

- [VI-626](https://jira.devops.va.gov/browse/VI-626)

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*
  **- The testing plan for rollout to to test the feature toggle locally (done), then on staging and then confirm on production that the feature toggle is effective.**
  
To test locally:
- [ ] In `config/settings.yml` set `create_after_login: true`
- [ ] Go to http://localhost:3000/flipper/features and find the new feature flag `mhv_account_creation_api`. Confirm it is enabled.
- [ ] Login to test if `create_mhv_account` is called, it should run `MHV::AccountCreatorJob.perform_async`
- [ ] Go back to http://localhost:3000/flipper/features and disable `mhv_account_creation_api`
- [ ] When disabled, we want to test if `create_mhv_account` is called upon logging in. It shouldn't.
- [ ] Optional, after disabling the feature flag we can edit the feature settings in the admin side of Flipper. You could play around with some of the settings and attempt more testing there.

## Screenshots
image
image

## What areas of the site does it impact?
- This impacts the `MHV::AccountCreatorJob` after a user logs in where the feature flag is enabled.

## Acceptance criteria
- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by bramleyjl — CHANGES_REQUESTED** [review]
Flipper feature worked for preventing SiS & SSOe authentications! I had some comments on the spec updates to simplify the spec cases between the YML setting and the Flipper setting.

Also, in your test instructions (and in local development generally) I would suggest updating the setting you want to change in `settings.local.yml` instead of modifying `settings.yml` directly, you can  add
```yml
mhv:
  account_creation:
    create_after_login: true
 ```
to help keep your local setup totally separate from the default that gets checked into source control.

File: spec/services/login/after_login_actions_spec.rb [hunk 1]
```diff
@@ -207,31 +207,58 @@
...
     end
 
-    context 'when creating an MHV account' do
+    context 'creating an MHV account' do
       let(:user) { create(:user, idme_uuid:) }
       let!(:user_verification) { create(:idme_user_verification, idme_uuid:) }
       let(:idme_uuid) { 'some-idme-uuid' }
 
-      context 'when mhv account_creation create_after_login is enabled' do
+      shared_examples 'and when mhv account_creation create_after_login is disabled' do
```
[the remark this answers is not in the document]
Comment (emilykim13) [reply spec/services/login/after_login_actions_spec.rb:215]: I was trying to not repeat myself too much. I can do this, it is actually what I had in mind originally

File: config/features.yml [hunk 2]
```diff
@@ -829,6 +829,10 @@ features:
     actor_type: user
     description: Enables notifications to be sent for new copay statements
     enable_in_development: true
+  mhv_account_creation_api:
```
Comment (rileyanderson) [inline config/features.yml:832]: we should name this something more descriptive like `mhv_account_creation_after_login`

File: spec/services/login/after_login_actions_spec.rb (hunk 1 above)
Comment (rileyanderson) [inline spec/services/login/after_login_actions_spec.rb:215]: `shared_examples` should primarily be used to describe shared behaviors and expectations across different contexts. They typically start with 'a' so when you use them it reads something like 
```ruby
it_behaves_like 'an action that doesn't enqueue a MHV::AccountCreatorJob' 
```
So you should leave this as a `context` then if you want shared_examples you could put them under the 'when creating an MHV account' context. Then the flow should be the overall setting enabled and disabled each test the flipper being enabled and disabled (again idk if we want to get rid of the setting but it's worth a discussion)

```ruby
context 'when creating an MHV account' do
  shared_examples 'an action that does not enqueue a MHV::AccountCreatorJob' do
    it 'does not enqueue a MHV::AccountCreatorJob' do
    end
  end

  context 'when the setting is enabled' do
    context 'when the flipper is enabled' do
      it 'enqueues an MHV::AccountCreatorJob' do
      end
      context 'when the flipper is disabled' do
        it_behaves like 'an action that does not enqueue a MHV::AccountCreatorJob'
      end
    end
    
    context 'when the setting is disabled' do
      context 'when the flipper is enabled' do
        it_behaves like 'an action that does not enqueue a MHV::AccountCreatorJob'
      end

      context 'when the flipper is disabled' do
        it_behaves like 'an action that does not enqueue a MHV::AccountCreatorJob'
      end
    end
  end
end
```
you _could_ also move some of the repeated contexts to `shared_contexts` but i don't think it's necessary

**Review by rileyanderson — CHANGES_REQUESTED** [review]

**Review by bramleyjl — APPROVED** [review]
Confirmed flipper functionality is properly protecting MHV account creation job call. Thanks for making those spec updates!

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**